source/fundamentals/type-hints.txt

```diff
@@ -1,340 +1,436 @@
+when you call the ``bulk_write()`` method:
 
 .. code-block:: python
 
-  >>> from typing import TypedDict
-  >>> from pymongo import MongoClient
-  >>> from pymongo.operations import InsertOne
-  >>> from pymongo.collection import Collection
-  >>> client: MongoClient = MongoClient()
-  >>> collection: Collection[Movie] = client.test.test
-  >>> inserted = collection.bulk_write([InsertOne(Movie(name="Jurassic Park", year=1993))])
-  >>> result = collection.find_one({"name": "Jurassic Park"})
-  >>> assert result is not None
-  >>> assert result["year"] == 1993
-  >>> # This will raise a type-checking error, despite being present, because it is added by PyMongo.
-  >>> assert result["_id"]  # type:ignore[typeddict-item]
+   >>> from typing import TypedDict
+   >>> from pymongo import MongoClient
+   >>> from pymongo.operations import InsertOne
+   >>> from pymongo.collection import Collection
+   >>> client: MongoClient = MongoClient()
+   >>> collection: Collection[Movie] = client.test.test
+   >>> inserted = collection.bulk_write([InsertOne(Movie(name="Jurassic Park", year=1993))])
+   >>> result = collection.find_one({"name": "Jurassic Park"})
+   >>> assert result is not None
+   >>> assert result["year"] == 1993
+   >>> # This will raise a type-checking error, despite being present, because it is added by PyMongo.
+   >>> assert result["_id"]  # type:ignore[typeddict-item]
 
 Modeling Document Types with TypedDict
 --------------------------------------
 
-You can use ``~typing.TypedDict`` (Python 3.8+) to model structured data.
-As noted above, PyMongo will automatically add an ``_id`` field if it is not present. This also applies to TypedDict.
-There are three approaches to this:
-
-1. Do not specify ``_id`` at all. It will be inserted automatically, and can be retrieved at run-time, but will yield a type-checking error unless explicitly ignored.
+All `schema validation <https://www.mongodb.com/docs/manual/core/schema-validation/#when-to-use-schema-validation>`__
+for inserts and updates is done on the server. These methods automatically add an ``_id``
+field to each document that doesn't include one. There are three ways to handle the ``_id``
+field in your custom ``TypedDict`` class.
 
-#. Specify ``_id`` explicitly. This will mean that every instance of your custom TypedDict class will have to pass a value for ``_id``.
+If you don't specify the ``_id`` field, {+driver-short+} automatically inserts it.
+You can retrieve the value of the field at run-time, but you'll see a type error
+at compile-time, as shown in the following example:
 
-#. Make use of ``~typing.NotRequired``. This has the flexibility of option 1, but with the ability to access the ``_id`` field without causing a type-checking error.
+If your code refers to the _id field, and your ``TypedDict`` doesn't
+contain this field, you will see a type error, as shown in the following example:
```

Comment:
It looks like you're introducing "the following example" in two different way, but saying the same thing. I think maybe the first intro works best?

Reply:
good catch